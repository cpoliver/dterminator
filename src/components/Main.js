import React from 'react';
import PropTypes from 'prop-types';
import Swiper from 'react-native-swiper';

import {
    Decider, Home, Profile, Randomizer, Settings
} from './pages';

const Main = ({ navigation }) => (
  <Swiper loop={false} showsPagination={false} index={1}>
    <Randomizer />
    <Swiper loop={false} showsPagination={false} index={1} horizontal={false}>
      <Profile />
      <Home />
      <Settings />
    </Swiper>
    <Decider navigation={navigation} />
  </Swiper>
);

Main.propTypes = {
  navigation: PropTypes.object.isRequired
};

export { Main };
