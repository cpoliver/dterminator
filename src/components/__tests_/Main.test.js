import 'react-native';
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { Main } from '../Main';

const render = () => shallow(<Main />);

describe('the Main component', () => {
  it('should match the snapshot', () => {
    const component = render();

    expect(toJson(component)).toMatchSnapshot();
  });
});
