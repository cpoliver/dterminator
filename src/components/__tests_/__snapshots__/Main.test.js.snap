// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the Main component should match the snapshot 1`] = `
<_class
  automaticallyAdjustContentInsets={false}
  autoplay={false}
  autoplayDirection={true}
  autoplayTimeout={2.5}
  bounces={false}
  horizontal={true}
  index={1}
  loadMinimal={false}
  loadMinimalSize={1}
  loop={false}
  pagingEnabled={true}
  removeClippedSubviews={true}
  scrollsToTop={false}
  showsButtons={false}
  showsHorizontalScrollIndicator={false}
  showsPagination={false}
  showsVerticalScrollIndicator={false}
>
  <Randomizer />
  <_class
    automaticallyAdjustContentInsets={false}
    autoplay={false}
    autoplayDirection={true}
    autoplayTimeout={2.5}
    bounces={false}
    horizontal={false}
    index={1}
    loadMinimal={false}
    loadMinimalSize={1}
    loop={false}
    pagingEnabled={true}
    removeClippedSubviews={true}
    scrollsToTop={false}
    showsButtons={false}
    showsHorizontalScrollIndicator={false}
    showsPagination={false}
    showsVerticalScrollIndicator={false}
  >
    <Profile />
    <Home />
    <Settings />
  </_class>
  <Connect(DecisionListComponent) />
</_class>
`;
